web/packages/teleport/src/BotInstances/List/BotInstancesList.tsx

```diff
@@ -16,148 +16,231 @@
-            <Cell>
-              <Flex inline alignItems={'center'} gap={1} mr={0}>
-                <MonoText>{instanceIdDisplay}</MonoText>
-                <CopyButton name={instance_id} />
-              </Flex>
-            </Cell>
-          ),
-        },
-        {
-          key: 'join_method_latest',
-          headerText: 'Method',
-          isSortable: false,
-          render: ({ join_method_latest }) =>
-            join_method_latest ? (
-              <LabelCell data={[join_method_latest]} />
-            ) : (
-              <Cell>{'-'}</Cell>
-            ),
-        },
-        {
-          key: 'host_name_latest',
-          headerText: 'Hostname',
-          isSortable: true,
-        },
-        {
-          key: 'version_latest',
-          headerText: 'Version (tbot)',
-          isSortable: true,
-        },
-        {
-          key: 'active_at_latest',
-          headerText: 'Last heartbeat',
-          isSortable: true,
-          render: ({ active_at_latest, activeAtLocal }) => (
-            <Cell>
-              <HoverTooltip tipContent={activeAtLocal}>
-                <span>{active_at_latest}</span>
-              </HoverTooltip>
-            </Cell>
-          ),
-        },
-      ]}
-      emptyText="No active instances found"
-      emptyButton={
-        <Info mt={5}>
-          Bot instances are ephemeral, and disappear once all issued credentials
-          have expired.
-        </Info>
-      }
-    />
+    <Container>
+      <TitleContainer>
+        <TitleText>Currently Active</TitleText>
+        <SortMenu
+          current={{
+            fieldName: sortField,
+            dir: sortDir,
+          }}
+          fields={sortFields}
+          onChange={value => {
+            onSortChanged(value.fieldName, value.dir);
+          }}
+        />
+      </TitleContainer>
+
+      <Divider />
+
+      {isLoading ? (
+        <Box data-testid="loading" textAlign="center" m={10}>
+          <Indicator />
+        </Box>
+      ) : undefined}
+
+      {hasError && hasUnsupportedSortError ? (
+        <Alert
+          m={3}
+          kind="warning"
+          primaryAction={{
+            content: 'Reset sort',
+            onClick: () => {
+              onSortChanged('bot_name', 'ASC');
+            },
+          }}
+        >
+          {error.message}
+        </Alert>
+      ) : undefined}
+
+      {hasError && !hasUnsupportedSortError ? (
+        <Alert m={3} kind="danger" details={error.message}>
+          Failed to fetch instances
+        </Alert>
+      ) : undefined}
+
+      {hasData ? (
+        <>
+          {data && data.length > 0 ? (
+            <ContentContainer ref={contentRef}>
+              {data.map((instance, i) => (
+                <React.Fragment key={`${instance.instance_id}`}>
+                  {i === 0 ? undefined : <Divider />}
+                  <Instance
+                    isSelectable
+                    onSelected={makeOnSelectedCallback(instance)}
```

Comment:
can instance be null? why do we need to have the option for it to be null in the onSelected